src/components/notebooks/modal.tsx

```diff
@@ -0,0 +1,64 @@
+import { useState } from 'react';
+import { XMarkIcon } from '@heroicons/react/24/solid';
+
+export default function Modal({
+  buttonTitle,
+  modalTitle,
+  buttonStyle,
+  children,
+}: {
+  buttonTitle: string;
+  modalTitle: string;
+  buttonStyle: string;
+  children: React.ReactNode;
+}) {
+  const [isModalOpen, setIsModalOpen] = useState(false);
+
+  const toggleModal = () => {
+    setIsModalOpen(!isModalOpen);
+  };
+
+  return (
+    <>
+      <button
+        type="button"
+        className={buttonStyle}
+        onClick={toggleModal}
+      >
+        {buttonTitle}
+      </button>
+
+      {isModalOpen && (
+        <div
+          id="authentication-modal"
+          tabIndex={-1}
+          aria-hidden="true"
+          className="fixed top-0 right-0 left-0 z-50 flex justify-center items-center h-screen"
+        >
+          <div className="relative p-4 w-full max-w-md max-h-full">
+            {/* Modal content */}
+            <div className="relative card rounded-lg shadow  border border-gray-400">
```

Comment:
Don't use card in this case, Create modal classes for this. Otherwise you'll get the properties of cards but it's not a card.